src/idp2p.ts

```diff
@@ -0,0 +1,163 @@
+import { Doctype } from '@ceramicnetwork/ceramic-common'
+import { DID } from 'dids'
+
+import { Accessors, createAccessors } from './accessors'
+import { IDX_DOCTYPE_CONFIGS, IdxDoctypeName } from './doctypes'
+import { AuthProvider, NewCeramicApi } from './types'
+
+export interface Idp2pOptions {
+  ceramic: NewCeramicApi
+}
+
+export interface AuthenticateOptions {
+  paths?: Array<string>
+  provider?: AuthProvider
+}
+
+export class Idp2p {
+  _ceramic: NewCeramicApi
+  _docQueue: Record<string, Promise<Doctype>> = {}
+  _rootDocId: string | undefined
+
+  accounts: Accessors
+  profiles: Accessors
+
+  constructor({ ceramic }: Idp2pOptions) {
+    this._ceramic = ceramic
+    this.accounts = createAccessors('accounts', this)
+    this.profiles = createAccessors('profiles', this)
+  }
+
+  get ceramic(): NewCeramicApi {
+    return this._ceramic
+  }
+
+  get user(): DID {
+    if (this._ceramic.user == null) {
+      throw new Error('User is not authenticated')
+    }
+    return this._ceramic.user
+  }
+
+  async authenticate(options: AuthenticateOptions = {}): Promise<void> {
+    let provider = this._ceramic.getDIDProvider()
```

Comment:
This is not a function.

Reply:
Let's just use it from opts for now.

Reply:
OK, yes I've been wondering about the flow here, is idp2p responsible for providing the provider to the Ceramic instance, or should it inject it?

I think we could still have a flow like:

```js
const provider = wallet.getDIDProvider()
ceramic.setDIDProvider(provider)
const idp2p = new Idp2p({ ceramic })
```

In that case I would expect idp2p to use the provider injected in Ceramic, no?

Reply:
Idp2p will use 3id-conenct by default (once it's ready). Regardless I think we should set the provider in the auth method (if passed).

Reply:
Yes the current logic sets the provider if the Ceramic instance doesn't already have one attached, but in order to detect this it needs to access it somehow. Or do you see another way this could be handled?

Reply:
Oh, so if the ceramic instance has a provider we don't do anything? I think that makes sense. Once js-did is in use in ceramic we can just check if the 
`ceramic.user` property is defined.

Reply:
If there is a provider the current logic is to check if there is a user, and create it otherwise, and always call the `user.authenticate()` method anyways, cf https://github.com/ceramicstudio/js-idp2p/pull/2/files#diff-8bce2c18125fb05b0778f7c9df7de809R56-R63

But if I understand correctly what you're writing, the Ceramic instance will always have a user if it has a provider, isn't it?
In that case indeed there's no need to access the provider here, only call `ceramic.setDIDProvider()` if `ceramic.user` isn't set and a provider is attached in this call.